Sources/XCTestCase+AnyCodableAsserts.swift

```diff
@@ -147,39 +249,146 @@ public protocol AnyCodableAsserts {
+    /// **and** have the same literal value.
+    ///
+    /// Both objects and arrays use extensible collections by default, meaning that only the elements in `expected` are
+    /// validated.
+    ///
+    /// Path options allow for powerful customizations to the comparison logic; see structs conforming to ``MultiPathConfig``:
+    /// - ``AnyOrderMatch``
+    /// - ``CollectionEqualCount``
+    /// - ``KeyMustBeAbsent``
+    /// - ``ValueExactMatch``, ``ValueTypeMatch``
+    ///
+    /// For example, given an expected JSON like:
+    /// ```
+    /// {
+    ///   "key1": "value1",
+    ///   "key2": [{ "nest1": 1}, {"nest2": 2}],
+    ///   "key3": { "key4": 1 },
+    ///   "key.name": 1,
+    ///   "key[123]": 1
+    /// }
+    /// ```
+    /// An example path for this JSON would be: `"key2[1].nest2"`.
+    ///
+    /// Paths must begin from the top level of the expected JSON. Multiple paths and path options can be used at the same time.
+    /// Path options are applied sequentially. If an option overrides an existing one, the overriding will occur in the order in which
+    /// the path options are specified.
+    ///
+    /// Formats for object keys:
+    /// - Standard keys - The key name itself: `"key1"`
+    /// - Nested keys - Use dot notation: `"key3.key4"`.
+    /// - Keys with dots in the name: Escape the dot notation with a backslash: `"key\.name"`.
+    ///
+    /// Formats for arrays:
+    /// - Standard index - The index integer inside square brackets: `[<INT>]` (e.g., `[0]`, `[28]`).
+    /// - Keys with array brackets in the name - Escape the brackets with backslashes: `key\[123\]`.
+    ///
+    /// Formats for wildcard object key and array index names:
+    /// - Array wildcard - All children elements of the array: `[*]` (ex: `key1[*].key3`)
+    /// - Object wildcard - All children elements of the object: `*` (ex: `key1.*.key3`)
+    /// - Key whose name is asterisk - Escape the asterisk with backslash: `"\*"`
+    /// - Note that wildcard path options also apply to any existing specific nodes at the same level.
+    ///
+    /// - Parameters:
+    ///   - expected: The expected `AnyCodableComparable` to compare.
+    ///   - actual: The actual `AnyCodableComparable` to compare.
+    ///   - pathOptions: The path options to use in the validation process.
+    ///   - file: The file from which the method is called, used for localized assertion failures.
+    ///   - line: The line from which the method is called, used for localized assertion failures.
     func assertExactMatch(expected: AnyCodableComparable, actual: AnyCodableComparable?, pathOptions: MultiPathConfig..., file: StaticString, line: UInt)
 }
 
 public extension AnyCodableAsserts where Self: XCTestCase {
-    // Exact equality is just a special case of exact match
+    /// Asserts exact equality between two `AnyCodableComparable` instances.
```

Comment:
nit: Others may have a different opinion, but you only need to document the protocol and not the concrete implementation, especially if they are in the same file. Sometimes I'll add a shorter doc to the implementation. However, it's fine to leave the documentation here since you've already included it.